pipenv/core.py

```diff
@@ -989,47 +1008,28 @@ def do_create_virtualenv(project, python=None, site_packages=None, pypi_mirror=N
         err=True,
     )
 
-    try:
-        import venv  # noqa
-
-        cmd = [
-            Path(sys.executable).absolute().as_posix(),
-            "-m",
-            "virtualenv",
-            "--creator=venv",
-            f"--prompt={project.name}",
-            f"--python={python}",
-            project.get_location_for_virtualenv(),
-        ]
-    except ImportError:
-        cmd = [
-            Path(sys.executable).absolute().as_posix(),
-            "-m",
-            "virtualenv",
-            f"--prompt={project.name}",
-            f"--python={python}",
-            project.get_location_for_virtualenv(),
-        ]
-
-    # Pass site-packages flag to virtualenv, if desired...
     if site_packages:
         click.echo(
             click.style(fix_utf8("Making site-packages available..."), bold=True),
             err=True,
         )
-        cmd.append("--system-site-packages")
 
     if pypi_mirror:
         pip_config = {"PIP_INDEX_URL": pypi_mirror}
     else:
         pip_config = {}
 
-    # Actually create the virtualenv.
     error = None
     with console.status(
         "Creating virtual environment...", spinner=project.s.PIPENV_SPINNER
     ):
-        c = subprocess_run(cmd, env=pip_config)
+        try:
+            cmd = _create_virtualenv_cmd(project, python, creator_venv=True, site_packages=site_packages)
+            c = subprocess_run(cmd, env=pip_config)
+        except (ImportError, FileNotFoundError, exceptions.VirtualenvCreationException):
```

Comment:
It seems to me this won't work since the underlying `subprocess_run` does not propagate those exceptions.

For example:
```
$ docker run --rm -it python:3.10 bash
...
root@c237ad213d85:/# python -m virtualenv
/usr/local/bin/python: No module named virtualenv
root@c237ad213d85:/# which pip
/usr/local/bin/pip
root@c237ad213d85:/# pip install virtualenv
...
root@c237ad213d85:/# python
Python 3.10.9 (main, Dec  8 2022, 01:35:40) [GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import virtualenv
>>> virtualenv.__file__
'/usr/local/lib/python3.10/site-packages/virtualenv/__init__.py'
>>> exit()
root@c237ad213d85:/# mv /usr/local/lib/python3.10/venv /usr/local/lib/python3.10/break_my_venv
root@c237ad213d85:/# python
Python 3.10.9 (main, Dec  8 2022, 01:35:40) [GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import subprocess as sp
>>> import sys
>>> sp.run(cmd, capture_output=True)
CompletedProcess(args=['/usr/local/bin/python', '-m', 'virtualenv', '--creator=venv', '--python=3.11', '/balls'], returncode=1, stdout=b"RuntimeError: failed to find interpreter for Builtin discover of python_spec='3.11'\n", stderr=b'')
```

Reply:
Try `c = sp.run(cmd, capture_output=True)` and `print int(c))`

Reply:
Printing c shows that there is exception but it's not raised.